src/main.rs

```diff
@@ -36,43 +36,44 @@ fn main() -> ExitCode {
     }
 
     // files
-    let errors: HashMap<String, Option<Diagnostic>> = args
-        .files
-        .into_iter()
-        .flat_map(|f| {
-            WalkDir::new(f)
-                .into_iter()
-                .filter_map(|e| e.ok())
-                .filter(|e| {
-                    e.file_name()
-                        .to_str()
-                        .map(|e| e.ends_with(".sql"))
-                        .unwrap_or(false)
-                })
-                .map(|f| -> (String, Option<Diagnostic>) {
-                    let key = f.into_path();
-                    (
-                        key.to_str().unwrap_or("").to_string(),
-                        fs::File::open(key)
-                            .map(|ref mut file| analyse_sql(file))
-                            .expect("failed to open file"),
-                    )
-                })
+    let targets = args.files.into_iter().flat_map(|f| {
+        WalkDir::new(f)
+            .into_iter()
+            .filter_map(|e| e.ok())
+            .filter(is_sql)
+    });
+    let errors = targets
+        .map(|f| -> (String, Option<Diagnostic>) {
+            let key = f.into_path();
+            (
+                key.to_str().unwrap_or("").to_string(),
+                fs::File::open(key)
+                    .map(|ref mut file| analyse_sql(file))
+                    .expect("failed to open file"),
+            )
         })
         .filter(|(_k, v)| v.is_some())
-        .collect();
+        .collect::<HashMap<String, Option<Diagnostic>>>();
+
     if !errors.is_empty() {
-        for (k, v) in errors.iter() {
-            if let Some(v) = v {
-                eprintln!("{}:{}", k, v);
+        for (fname, diag) in errors.iter() {
+            if let Some(diag) = diag {
+                eprintln!("{}:{}", fname, diag);
             }
         }
         return ExitCode::FAILURE;
     }
-
     ExitCode::SUCCESS
 }
 
+fn is_sql(entry: &DirEntry) -> bool {
+    entry
+        .file_name()
+        .to_str()
+        .map(|s| s.ends_with(".sql"))
+        .unwrap_or(false)
+}
+
```

Comment:
This code borrowed from below

https://docs.rs/walkdir/latest/walkdir/#example-skip-hidden-files-and-directories-on-unix